### PR TITLE
implement metrics as losses

### PR DESCRIPTION
**Pull request by vict0rsch**
addresses #20

### CODE REVIEW

File: aiphysim/metrics.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import traceback as tb
+
+import torch
+
+
+class BaseMetrics:
+    def __init__(self, opts):
+        self.metrics_funcs = {
+            loss_name: getattr(BaseMetrics, loss_name) for loss_name in opts.losses
```
Comment (jaggbow) [inline aiphysim/metrics.py:9]: Shouldn't this correspond to opt.metrics and we add metrics in the opt?

**Review by jaggbow — COMMENTED** [review]
I added a comment for the metrics.py file

File: aiphysim/metrics.py (hunk 1 above)
Comment (vict0rsch) [reply aiphysim/metrics.py:9]: yes, typo!

Comment (sm7610) [reply aiphysim/metrics.py:9]: How about adding it as a type of "loss" in opts, and use "loss_type = spacetime_metrics"? That seems cleaner to me.

Comment (vict0rsch) [reply aiphysim/metrics.py:9]: I think metrics and losses are conceptually  2 different things and we might want to separate their logics. I'm not even sure how this would look like in the code since they would not be called in the same locations of the code
